drivers/windows/file_access_windows.cpp

```diff
@@ -130,10 +130,16 @@ Error FileAccessWindows::open_internal(const String &p_path, int p_mode_flags) {
 
 	if (is_backup_save_enabled() && p_mode_flags == WRITE) {
 		save_path = path;
-		path = path + ".tmp";
+		// Create a temporary file in the same directory as the target file.
+		WCHAR tmpFileName[MAX_PATH];
+		if (GetTempFileNameW((LPCWSTR)(path.get_base_dir().utf16().get_data()), (LPCWSTR)(path.get_file().utf16().get_data()), 0, tmpFileName) == 0) {
+			last_error = ERR_FILE_CANT_OPEN;
+			return last_error;
+		}
+		path = tmpFileName;
```

Comment:
```suggestion
		path = String::utf16(tmpFileName);
```

Windows strings are UTF-16, Godot strings are UTF-32, so conversion is necessary.

Reply:
The String assignment operator does this implicitly as far as I can tell:

https://github.com/godotengine/godot/blob/27b2260460ab478707d884a16429add5bb3375f1/core/string/ustring.cpp#L463

https://github.com/godotengine/godot/blob/27b2260460ab478707d884a16429add5bb3375f1/core/string/ustring.cpp#L352